File: TransCoder_tokenized_test_set_functions/python/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.py
def check ( st ) :
    n = len ( st )
    if ( n == 0 ) :
        return False
    if ( n == 1 ) :
        return ( ( st [ 0 ] - '0' ) % 4 == 0 )
    last = ( int ) ( st [ n - 1 ] )
    second_last = ( int ) ( st [ n - 2 ] )
    return ( ( second_last * 10 + last ) % 4 == 0 )

File: TransCoder_tokenized_test_set_functions/java/CHECK_LARGE_NUMBER_DIVISIBLE_4_NOT.java
static boolean check ( String str ) {
  int n = str . length ( );
  if ( n == 0 ) return false;
  if ( n == 1 ) return ( ( str . charAt ( 0 ) - '0' ) % 4 == 0 );
  int last = str . charAt ( n - 1 ) - '0';
  int second_last = str . charAt ( n - 2 ) - '0';
  return ( ( second_last * 10 + last ) % 4 == 0 );
}